src/main/java/com/google/sps/model/GeocodingClient.java

```diff
@@ -0,0 +1,87 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.model;
+
+import com.google.maps.model.AddressType;
+import com.google.maps.model.GeocodingResult;
+import com.google.maps.model.LatLng;
+import com.google.maps.model.PlaceType;
+import com.google.sps.exceptions.GeocodingException;
+
+/**
+ * Contract for sending GET requests to the Google Geocoding API. Implement getGeocodingResult to
+ * obtain the corresponding GeocodingResult of an address.
+ */
+public interface GeocodingClient {
+  /**
+   * Sends a GET request to the Google Geocoding API to convert from address to GeocodingResult.
+   *
+   * @param address A String representing the address to geocode.
+   * @return A GeocodingResult returned from the Geocoding API.
+   * @throws GeocodingException A custom exception is thrown to signal an error pertaining to the
+   *     Geocoding API.
+   */
+  GeocodingResult getGeocodingResult(String address) throws GeocodingException;
+
+  /**
+   * Parses for coordinates in a resulting call to the Geocoding API.
+   *
+   * @param result A GeocodingResult returned from the Geocoding API.
+   * @return A LatLng representing coordinates.
+   */
+  public static LatLng getCoordinates(GeocodingResult result) {
```

Comment:
nit: `public` is redundant in interfaces

Reply:
I keep forgetting this - thanks!

Reply:
All these static methods should live in a `GeocodingResultUtil` or similarly named class. Then the tests would live in a more appropriately-named test class. (These methods have nothing to do with the client, and you don't actually have any tests for the implementation of the client.)

Reply:
Agreed, I should probably do the same for a lot of the PRs I have since I put most of the methods into the clients as long as they're remotely related (i.e. uses the same API). Test names would make a lot more sense as well since before the changes they're all in GeocodingClientImplTest - which makes no sense.

Reply:
`RESOLVED`